src/json.hpp.re2c

```diff
@@ -8097,64 +8097,143 @@ class basic_json
             return result;
         }
 
-        /*!
-        @brief parse floating point number
+        // non locale aware isdigit
+        // Microsoft in 1252 codepage and others may classify additional single-byte characters as digits using std::isdigit
+        constexpr bool nl_isdigit(const char c) const
+        {
+            return c >= '0' and c <= '9';
+        }
 
-        This function (and its overloads) serves to select the most approprate
-        standard floating point number parsing function based on the type
-        supplied via the first parameter.  Set this to @a
-        static_cast<number_float_t*>(nullptr).
+        /*!
+        @brief parse string to floating point number
 
-        @param[in] type  the @ref number_float_t in use
+        This function is a partial reimplementation of the strtold in order to meet needs of JSON number
 
-        @param[in,out] endptr recieves a pointer to the first character after
-        the number
+        @param[in] str  the string we will parse
 
         @return the floating point number
         */
-        long double str_to_float_t(long double* /* type */, char** endptr) const
+        long double strtojnum(const char *str) const
         {
-            return std::strtold(reinterpret_cast<typename string_t::const_pointer>(m_start), endptr);
-        }
+            long double result = 0;
+            char cp = *str;
+            int exp = 0; // exponent
+            {
+                const bool negative_sign = cp == '-';
 
-        /*!
-        @brief parse floating point number
+                if (cp == '-' or cp == '+')
+                {
+                    ++str;
+                }
 
-        This function (and its overloads) serves to select the most approprate
-        standard floating point number parsing function based on the type
-        supplied via the first parameter.  Set this to @a
-        static_cast<number_float_t*>(nullptr).
+                // read in fractional part of number, until an 'e' is reached.
+                // count digits after decimal point.
+                while (nl_isdigit(cp = *str))
+                {
+                     result = result * 10 + (cp - '0');
+                     ++str;
```

Comment:
Couldn't `str` be incremented in the `while` condition just like in the loop below?